Tutorials/Members-Registration-And-Logins/index-v10.md

```diff
@@ -0,0 +1,360 @@
+            catch (Exception ex)
+            {
+                //handle the exception
+            }
+
+        }
+
+
+        /// <summary>
+    
+        /// </summary>
+        /// <param name="model">Register member model.</param>
+        /// <param name="logMemberIn">Flag for whether to log the member in upon successful registration.</param>
+        /// <returns>Result of registration operation.</returns>
+        private async Task<IdentityResult> RegisterMemberAsync(RegisterModel model, bool logMemberIn = true)
+        {
+            using ICoreScope scope = _scopeProvider.CreateCoreScope(autoComplete: true);
+
+           
+            if (string.IsNullOrEmpty(model.Name) && string.IsNullOrEmpty(model.Email) == false)
+            {
+                model.Name = model.Email;
+            }
+
+            model.Username = model.UsernameIsEmail || model.Username == null ? model.Email : model.Username;
+
+            var identityUser =
+                MemberIdentityUser.CreateNew(model.Username, model.Email, model.MemberTypeAlias, true, model.Name);
+            IdentityResult identityResult = await _memberManager.CreateAsync(
+                identityUser,
+                model.Password);
+
+            if (identityResult.Succeeded)
+            {
+               
+                IMember? member = _memberService.GetByKey(identityUser.Key);
+                if (member == null)
+                {
+                    
+                    throw new InvalidOperationException($"Could not find a member with key: {member?.Key}.");
+                }
+
+                foreach (MemberPropertyModel property in model.MemberProperties.Where(p => p.Value != null).Where(property => member.Properties.Contains(property.Alias)))
+                {
+                    member.Properties[property.Alias]?.SetValue(property.Value);
+                }
+
+                //Before we save the member we make sure to assigne the group, for this the "Group" must exist in the backoffice.
+                string memberGroup = "professionals";
+                AssignMemberGroup(model.Email, memberGroup);
+
+                _memberService.Save(member);
+
+                if (logMemberIn)
+                {
+                    await _memberSignInManager.SignInAsync(identityUser, false);
+                }
+            }
+
+            return identityResult;
+        }
+    }
+}
+```
+
+For an easier implementation, you can copy the above code to a new .cs file and place it in a folder in your solution so that it will be compiled on application startup.
+
+With the above controller in place it is time to adjust the macro/view files as well.
+
+:::note
+In the example above we have only hardcoded the Member group we are assigning to someone who are registrating in our example.
+
+To be able to add the Member Group that we automaticly assign from the backoffice you will need to extend the controller further, which we won't cover in this tutorial.
+:::
```

Comment:
```suggestion
For an easier implementation, you can copy the above code to a new .cs file and place it in a folder in your solution so that it will be compiled on application startup.

With the above controller in place, it is time to adjust the macro/view files as well.

:::note
In the example above we have only hardcoded the Member group, which we are assigning to someone who is registering in our example.

To be able to add the Member Group that we automatically assign from the backoffice you will need to extend the controller further, which we won't cover in this tutorial.
:::
```